add docs for subscription APIs (#1698)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/admin-notifications/index.md

| notificationType | `type` of a notification                               | URL         | [Notification Type](#notification-type)     | TRUE      |

### Response parameters

[Subscription](#subscription-object)

### Request example

```bash
curl -v -X GET \
-H "Accept: application/json" \
-H "Content-Type: application/json" \
-H "Authorization: SSWS ${api_token}" \
"https://${yourOktaDomain}/api/v1/roles/SUPER_ADMIN/subscriptions/IWA_AGENT"
```

### Response example

```json
{
    "notificationType": "IWA_AGENT",
    "channels": [
        "email"
    ],
    "status": "subscribed",
    "_links": {
        "unsubscribe": {
            "href": "https://${yourOktaDomain}/api/v1/roles/SUPER_ADMIN/subscriptions/IWA_AGENT/unsubscribe",
            "hints": {
                "allow": [
                    "POST"
                ]
            }
        },
        "self": {
            "href": "https://${yourOktaDomain}/api/v1/roles/SUPER_ADMIN/subscriptions/IWA_AGENT"
        }
    }
}
```

## Subscribe to a specific type

<ApiOperation method="post" url="/api/v1/roles/${roleType}/subscriptions/${notificationType}/subscribe" />

> Subscribes a Role to a specific type. When you change the subscription status of a role, it will override the subscription of any individual user of that role.

Reviewer: ```suggestion
Subscribes a Role to a specific type. When you change the subscription status of a Role, it overrides the subscription of any individual user of that Role.
```

### Request parameters

| Parameter        | Description                                            | Param Type  | DataType                                 | Required  |
| :--------------- | :----------------------------------------------------- | :---------- | :--------------------------------------- | :-------- |
| roleType         | `type` of a Role                                       | URL         | [Role Type](#role-types)                 | TRUE      |
| notificationType | `type` of a notification                               | URL         | [Notification Type](#notification-type)  | TRUE      |

### Response parameters

``` http
HTTP/1.1 200 OK
```

### Request example

```bash
curl -v -X POST \
-H "Accept: application/json" \
-H "Content-Type: application/json" \
-H "Authorization: SSWS ${api_token}" \
"https://${yourOktaDomain}/api/v1/roles/SUPER_ADMIN/subscriptions/IWA_AGENT/subscribe"
```

### Response example

``` http
HTTP/1.1 200 OK
```

## Unsubscribe to a specific type

<ApiOperation method="post" url="/api/v1/roles/${roleType}/subscriptions/${notificationType}/unsubscribe" />

> Unsubscribes a Role to a specific type. When you change the subscription status of a role, it will override the subscription of any individual user of that role.

Reviewer: ```suggestion
Unsubscribes a Role to a specific type. When you change the subscription status of a Role, it overrides the subscription of any individual user of that Role.
```

### Request parameters

| Parameter        | Description                                            | Param Type  | DataType                                 | Required  |
| :--------------- | :----------------------------------------------------- | :---------- | :--------------------------------------- | :-------- |
| roleType         | `type` of a Role                                       | URL         | [Role Type](#role-types)                 | TRUE      |
| notificationType | `type` of a notification                               | URL         | [Notification Type](#notification-type)  | TRUE      |

### Response parameters

``` http
HTTP/1.1 200 OK
```

### Request example

```bash
curl -v -X POST \
-H "Accept: application/json" \
-H "Content-Type: application/json" \
-H "Authorization: SSWS ${api_token}" \
"https://${yourOktaDomain}/api/v1/roles/SUPER_ADMIN/subscriptions/IWA_AGENT/unsubscribe"
```

### Response example

``` http
HTTP/1.1 200 OK
```

## List subscriptions of a User

<ApiOperation method="get" url="/api/v1/users/${userId}/subscriptions" />

> List subscriptions of a User. We only support operations for current user right now. Requesting for other users will get AccessDeniedException.

Reviewer: ```suggestion
List subscriptions of a User. Only lists subscriptions for current users. An AccessDeniedException generates if requesting from other users.
```

                "hints": {
                    "allow": [
                        "POST"
                    ]
                }
            },
            "self": {
                "href": "https://${yourOktaDomain}/api/v1/users/00uuk0UVgUXjkIbPL0g3/subscriptions/USER_DEPROVISION"
            }
        }
    },
    {
        "notificationType": "REPORT_SUSPICIOUS_ACTIVITY",
        "channels": [
            "email"
        ],
        "status": "subscribed",
        "_links": {
            "unsubscribe": {
                "href": "https://${yourOktaDomain}/api/v1/users/00uuk0UVgUXjkIbPL0g3/subscriptions/REPORT_SUSPICIOUS_ACTIVITY/unsubscribe",
                "hints": {
                    "allow": [
                        "POST"
                    ]
                }
            },
            "self": {
                "href": "https://${yourOktaDomain}/api/v1/users/00uuk0UVgUXjkIbPL0g3/subscriptions/REPORT_SUSPICIOUS_ACTIVITY"
            }
        }
    }
]
```

## Get the subscription of a User with a specific type

<ApiOperation method="get" url="/api/v1/roles/${userId}/subscriptions/${notificationType}" />

> Get the subscriptions of a User with a specific type. We only support operations for current user right now. Requesting for other users will get AccessDeniedException.

Reviewer: ```suggestion
Get the subscriptions of a User with a specific type. Only lists subscriptions for current user. An AccessDeniedException generates if requesting from other users.
```

| notificationType | `type` of notification | URL         | [Notification Type](#notification-type)     | TRUE      |

### Response parameters

[Subscription](#subscription-object)

### Request example

```bash
curl -v -X GET \
-H "Accept: application/json" \
-H "Content-Type: application/json" \
-H "Authorization: SSWS ${api_token}" \
"https://${yourOktaDomain}/api/v1/users/00uuk0UVgUXjkIbPL0g3/subscriptions/IWA_AGENT"
```

### Response example

```json
{
    "notificationType": "IWA_AGENT",
    "channels": [
        "email"
    ],
    "status": "subscribed",
    "_links": {
        "unsubscribe": {
            "href": "https://${yourOktaDomain}/api/v1/users/00uuk0UVgUXjkIbPL0g3/subscriptions/IWA_AGENT/unsubscribe",
            "hints": {
                "allow": [
                    "POST"
                ]
            }
        },
        "self": {
            "href": "https://${yourOktaDomain}/api/v1/users/00uuk0UVgUXjkIbPL0g3/subscriptions/IWA_AGENT"
        }
    }
}
```

## Subscribe to a specific type

<ApiOperation method="post" url="/api/v1/users/${userId}/subscriptions/${notificationType}/subscribe" />

> Subscribes a User to a specific type. We only support operations for current user right now. Requesting for other users will get AccessDeniedException.

Reviewer: ```suggestion
Subscribes a User to a specific type. Only lists subscriptions for current user. An AccessDeniedException generates if requesting from other users.
```

### Request parameters

| Parameter        | Description             | Param Type  | DataType                                 | Required  |
| :--------------- | :---------------------  | :---------- | :--------------------------------------- | :-------- |
| userId           | `id` of a User          | URL         | String                                   | TRUE      |
| notificationType | `type` of notification  | URL         | [Notification Type](#notification-type)  | TRUE      |

### Response parameters

``` http
HTTP/1.1 200 OK
```

### Request example

```bash
curl -v -X POST \
-H "Accept: application/json" \
-H "Content-Type: application/json" \
-H "Authorization: SSWS ${api_token}" \
"https://${yourOktaDomain}/api/v1/users/00uuk0UVgUXjkIbPL0g3/subscriptions/IWA_AGENT/subscribe"
```

### Response example

``` http
HTTP/1.1 200 OK
```

## Unsubscribe to a specific type

<ApiOperation method="post" url="/api/v1/users/${userId}/subscriptions/${notificationType}/unsubscribe" />

> Unsubscribes a User to a specific type. We only support operations for current user right now. Requesting for other users will get AccessDeniedException.

Reviewer: ```suggestion
Unsubscribes a User to a specific type. Only unsubscribes subscriptions for current user. An AccessDeniedException generates if requesting from other users.
```

```json
{
    "notificationType": "IWA_AGENT",
    "channels": [
        "email"
    ],
    "status": "subscribed",
    "_links": {
        "unsubscribe": {
            "href": "https://${yourOktaDomain}/api/v1/users/00uuk0UVgUXjkIbPL0g3/subscriptions/IWA_AGENT/unsubscribe",
            "hints": {
                "allow": [
                    "POST"
                ]
            }
        },
        "self": {
            "href": "https://${yourOktaDomain}/api/v1/users/00uuk0UVgUXjkIbPL0g3/subscriptions/IWA_AGENT"
        }
    }
}
```

### Subscription properties

The Subscription object defines several properties:

| Property         | Description                                                  | DataType                                                        | Nullable   | Unique   | Read Only |
| :--------------- | :----------------------------------------------------------- | :-------------------------------------------------------------- | :--------- | :------- | :-------- |
| notificationType | Type of subscriptions                                        | [Notification Type](#notification-type)                         | FALSE      | TRUE     | TRUE      |

> | channels         | Array of sources that the user will get notification from    | array of [Channel](#subscription-channel)                       | FALSE      | FALSE    | TRUE      |

Reviewer: ```suggestion
| channels         | Array of sources that the user gets notification from    | array of [Channel](#subscription-channel)                       | FALSE      | FALSE    | TRUE      |
```

| AD_AGENT                   | System notification sent when an AD agent disconnects or reconnects.                                           |
| OKTA_ANNOUNCEMENT          | Okta communication sent for announcements and release notes.                                                   |
| OKTA_ISSUE                 | Okta communication sent for trust incident alerts and updates.                                                 |
| OKTA_UPDATE                | Okta communication sent for scheduled system updates.                                                          |
| IWA_AGENT                  | System notification sent when an IGA agent disconnects or reconnects.                                          |
| USER_DEPROVISION           | System notification sent when a user is deprovisioned from apps.                                               |
| REPORT_SUSPICIOUS_ACTIVITY | System notification sent when a user reports suspicious activity.                                              |
| RATELIMIT_NOTIFICATION     | System notification sent when an org reaches rate limit warning or violation thresholds.                       |

## Role types

> Some Roles support optional targets that constrain the Role to a specific set of Groups or Apps. If an optional target isn't specified, then the role assignment is unbounded (for example, applies to all Groups or Apps).

Reviewer: ```suggestion
Some Roles support optional targets that constrain the Role to a specific set of Groups or Apps. If an optional target isn't specified, then the Role assignment is unbounded (for example, applies to all Groups or Apps).
```
